src/main/java/org/hyperledger/aries/webhook/WebSocketListener.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020-2022 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/acapy-java-client
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.aries.webhook;
+
+import org.hyperledger.aries.config.GsonConfig;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+
+@Slf4j
+public class WebSocketListener extends okhttp3.WebSocketListener {
+
+    private Gson gson = GsonConfig.defaultConfig();
+    
+    private final String label;
+    private final EventHandler handler;
+    
+    public WebSocketListener(String label, EventHandler handler) {
```

Comment:
A comment would be nice describing the purpose of the label. Also it should be optional I guess. Also the constructor should be annotated with `@Builder`